buffer/src/main/java/io/netty/buffer/CompositeByteBuf.java

```diff
@@ -1333,7 +1333,7 @@ public int setBytes(int index, FileChannel in, long position, int length) throws
     @Override
     public ByteBuf copy(int index, int length) {
         checkIndex(index, length);
-        ByteBuf dst = Unpooled.buffer(length);
+        ByteBuf dst = alloc().buffer(length);
```

Comment:
Maybe `allocBuffer(length)`?

Reply:
@fenik17 good catch... let me do this and add one more assert in the tests 👍